File: 1-two-sum/two-sum.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        (int num, int index)[] numbers = nums.Select((num, index) => (num, index)).ToArray();
        Array.Sort(numbers, (a, b) => a.num.CompareTo(b.num));
        int i = 0, j = nums.Length - 1;
        
        while (i < j) {
            int sum = numbers[i].num + numbers[j].num;
            if (sum == target) {
                return new int[] { numbers[i].index, numbers[j].index };
            } else if (sum < target) {
                i++;
            } else {
                j--;
            }
        }
        
        throw new ArgumentException("No two sum solution");
    }
}

File: 1-two-sum/two-sum.dart
class Solution {

List<int> twoSum(List<int> nums, int target) {
  Map<int, dynamic> numMap = {};

  for (int i = 0; i < nums.length; i++) {
    int complement = target - nums[i];
    if (numMap.containsKey(complement)) {
      return [numMap[complement], i];
    }
    numMap[nums[i]] = i;
  }

  return [];
}
}